server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -329,7 +331,7 @@ public IndexShard(
         final RetentionLeaseSyncer retentionLeaseSyncer,
         final CircuitBreakerService circuitBreakerService,
         @Nullable final SegmentReplicationCheckpointPublisher checkpointPublisher,
-        @Nullable final RemoteStoreRefreshListener remoteStoreRefreshListener
+        @Nullable final Store remoteStore
```

Comment:
I'm not sure if the `Store` should be abstracted as a `CompositeStore` that has data backed in remote. IndexShard shouldn't worry about exposing or maintaining another `Store` that is remote

Reply:
Need some time to think about this abstraction. Tracking here: https://github.com/opensearch-project/OpenSearch/issues/3719